### ISSUE TITLE
my profile just deleted itself

### ISSUE THREAD

**Issue by Tcll** [uTox/uTox#1357]
### Overview

1. uTox version: 0.17.0
2. Operating system: Void Linux
3. Desktop Environment/Window Manager (Linux/BSD only): XFCE+lightdm

### Issue

was just randomly typing a message to my friend when my client froze, so I killed it's process...
after reloading the client, I had a completely new profile.
so I went browsing through my directories in my .config/ folder and found my tox/ folder was missing.

if I hadn't had a backup, I'd've lost 19 friends
thankfully I'd been keeping backups since utox 0.16.1 corrupted my last profile to where the password was always invalid, but still, this is a major issue that needs to be addressed.

**Comment by avoidr** [comment]
Oh my, I'm really sorry this happened! I was really worried but luckily you had a backup! Phew 😥

Do you by any chance have any hint as to how to reproduce this freeze? Did you notice anything else unusual on the system at the time it happened?
Are you running the git version of utox?

**Comment by Tcll** [comment]
can't say I'm sure as to what causes it, I haven't really launched utox through a terminal to get any debug info
should probably note I crashed my friend's client as well (or maybe he froze me when he crashed)
he uses 0.16.1 for windows from the old site

but anyways, for some things you can probably experiment with:
I'm running on a cruddy hyperthreaded Pentium4 with only 3.5GB of memory
I'm also running Minecraft on it connected to my server
but really the point of that is just to point out my CPU is very busy, and my client tends to freeze often as a result...
swap use may also contribute to things

but yeah I'm not sure if those are the leading factors
would there be a log anywhere I could look into on this??

**Comment by avoidr** [comment]
Hmm. You could check if you notice anything in your `dmesg`.
And it would have been useful if you had a coredump. You would have to enable it, though; `ulimit -c unlimited`

**Comment by Tcll** [comment]
ah thanks
yeah it was a segfault
error 4 in libc-2.29.so[7f8cde34c000+149000]

code

    0f 1f 40 00 66 0f ef c0 66 0f ef c9 66 0f ef d2 66 0f ef db 48 89 f8 48 89 f9 48 81 e1 ff 0f 00 00 48 81 f9 cf 0f 00 00 tt 6a <f3> 0f 6f 20 66 0f 74 e0 66 0f d7 d4 85 d2 74 04 0f bc c2 c3 48 83

if you need more, I can get
I had to type everything in because my browser is open on another machine :P

**Comment by avoidr** [comment]
Did you use ctrl+z in your message field, maybe? Do you ever use it? It's crashy.

So I guess you're using the release versions of utox, not compiled yourself, right?

You're a Void Linux user, you'll know how to enable core dumps :P

**Comment by avoidr** [comment]
What address does it tell at "segfault at"? Is it 0?

**Comment by Tcll** [comment]
just because I switched to Arch from Ubuntu doesn't mean I know everything (core dumps being one of those things) X3
I'll look it up though :P
But yeah, I'm not interested in building my own distro (I got other crap to work on), I'm just interested in having something secure out of the box :)

and yeah, just from xbps, which I was actually surprised had a release
I don't prefer to build from source if I don't absolutely have to (same for the terminal really)

anyways, no, I'm not even sure ctrl+z works for me (I know it didn't on 0.16)

and the address was on [303039.204591]
wasn't sure if that was needed, but yeah

**Comment by avoidr** [comment]
>and the address was on [303039.204591]

No, that's the time :) I want the number that comes after "segfault at".

>I don't prefer to build from source if I don't absolutely have to (same for the terminal really)

Maybe that explains why ulimit wasn't found? ulimit is a shell builtin.

I would suggest to use utox from git. The latest release is over a year back and there were a few changes since then.

**Comment by avoidr** [comment]
Core dumps are not a systemd feature. Though systemd probably "collects" them for you in some "convenient" way. Though, I thought you were using Void Linux? Which doesn't use systemd.

**Comment by endoffile78** [comment]
Systemd has coredumpctl. Using coredumpctl dump <pid> you can get the coredump

**Comment by Tcll** [comment]
Void Linux is an Arch-based distro just like Xubuntu is an Ubuntu-based distro
that's what I was referring to there :)

anyways, so that's what that was
sorry, I'm used to python where everything in the middle is not really needed XP
(shoulda just typed the full line)
C is just on my TODO list :P

anyways:

    segfault at 1 ip 0007f8cde3c1826 sp 00007ffe521ff5f8 error 4 ...

**Comment by avoidr** [comment]
>Void Linux is an Arch-based distro

lol, no.

**Comment by Tcll** [comment]
> lol, no.
I see, I can accept being wrong here
I forget where I read it was arch-based, but after doing a search, I see it's better than arch from this notable source:
https://itsfoss.com/void-linux/

in any case, I couldn't seem to find any relevant documentation for enabling core dumps, so I've asked here:
https://unix.stackexchange.com/questions/518659/how-do-i-enable-core-dumps-in-void-linux
you guys can answer if you know how, it would certainly help others trying to figure things out. ;)

anyways, I think this issue will have to happen again once I have core-dumps going, and have built the git version, so I think this'll have to be on hold for now, unless my earlier info told you roughly/exactly what was causing the issue.

but on a final note, if I'm gonna be building it...
is it possible to build a portable version I can run on a flash drive, or copy across multiple machines for various purposes??
(my typical/preferred use case is an installation on another HDD in case I fix/change/add distros)

**Comment by avoidr** [comment]
Well, you could have just checked Void Linux' homepage :)

I've already said how to enable core dumps. `ulimit -c unlimited`. Bash and dash have this shell builtin. Then you start utox from the same shell.

>is it possible to build a portable version I can run on a flash drive, or copy across multiple machines for various purposes??

What you want is static linking.
